Content.Trauma.Server/Botany/Components/PlantAnalyzerComponent.cs

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: 2025 Liamofthesky <157073227+Liamofthesky@users.noreply.github.com>
+// SPDX-FileCopyrightText: 2025 taydeo <td12233a@gmail.com>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later AND MIT
+
+using System.Linq;
+using Content.Shared.Atmos;
+using Content.Trauma.Server.Botany;
+using Content.Trauma.Shared.Botany.Components;
+using Content.Shared.DoAfter;
+using Robust.Shared.Audio;
+
+namespace Content.Trauma.Server.Botany.Components;
+
+/// <summary>
+///    After scanning, retrieves the target Uid to use with its related UI.
+/// </summary>
+[RegisterComponent]
+public sealed partial class PlantAnalyzerComponent : Component
+{
+    [DataDefinition]
+    public partial struct PlantAnalyzerSetting
+    {
+        [DataField]
+        public PlantAnalyzerModes AnalyzerModes;
+
+        [DataField]
+        public float ScanDelay;
+
+        [DataField]
+        public float ModeDelay;
+    }
```

Comment:
use DataRecord instead of DataDefinition since its all just DataField and remove the DataFields

also move the struct outside of the component so its not awful to reference